### ISSUE TITLE
http_listener service plugin does not respect precision

### ISSUE THREAD

**Issue by kostasb** [influxdata/telegraf#2637]
Telegraf v1.2.1

Issue: the http_listener plugin does not accept configurable precision, only assumes ns.
Also, outputs.influxdb only writes ns.

The result is that points with a precision other than ns writen through inputs.http_listener land in the wrong time.

Repro case:

-Setup a telegraf agent applying the following config:

```
[[inputs.http_listener]]
   service_address = ":8186"
   read_timeout = "10s"
   write_timeout = "10s"
   max_body_size = 0
   max_line_size = 0

[outputs.influxdb]]
  urls = ["http://localhost:8086"] # required
  database = "telegraf" # required
  retention_policy = ""
  write_consistency = "any"
  timeout = "5s"
```

-Submit a point with precision other than ns to Telegraf's http input listener:

```
curl -v "http://localhost:8186/write?db=test&precision=ms" --data-binary "test value=1i 1491601527552"
```

-Check the timestamp written for this point in InfluxDB:

```
> select * from test
name: test
time                           host   value
----                           ----   -----
1970-01-01T00:24:51.601527552Z ubuntu 1
```

**Comment by kostasb** [comment]
Related to and subset of: https://github.com/influxdata/telegraf/issues/1589

**Comment by danielnelson** [comment]
Found this as well https://github.com/influxdata/telegraf/issues/2359

I don't agree though, http_listener is very specific to InfluxDB so I see no reason not to support precision.

**Comment by danielnelson** [comment]
We don't support the db parameter here either, this probably shouldn't change because the output defines it.

**Comment by danielnelson** [comment]
Implemented in #2644
